### ISSUE TITLE
Segmentation fault while calling InputSource-Constructor

### ISSUE THREAD

**Issue by githubmeier** [pocoproject/poco#3148]
We are facing a random segmentation fault issue while calling the following code:

std::string fileName = "xmlfile.xml";
std::ifstream in(fileName);
Poco::XML::InputSource src(in);     // The crash occurs here...

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff4ac6d97 in std::operator<< <char, std::char_traits<char>, std::allocator<char> > (__os=..., __str=...)
    at /usr/src/debug/gcc10-10.2.1+git583-lp152.2.2.x86_64/obj-x86_64-suse-linux/x86_64-suse-linux/libstdc++-v3/include/bits/basic_string.h:3903
3903	      size() const _GLIBCXX_NOEXCEPT

Any ideas or further investigations?

**Comment by Mironenko** [comment]
Is there any minimal sample to reproduce (in cycle, till segfault, for instance)?

**Comment by githubmeier** [comment]
@Mironenko : Thank you for your reply. I think an example is not necessary anymore.

Meanwhile we were able to solve the issue, as we tried different XML, different reading possibilities and so on...

The pre-compiled library in opensuse is compiled with standard ABI-flags, where as we compile our application with -D_GLIBCXX_USE_CXX11_ABI=0.

We downloaded a current version of POCO and compiled it also with -D_GLIBCXX_USE_CXX11_ABI=0 ... which works :-)
